pkg/runtime/reconciler.go

```diff
@@ -166,36 +162,48 @@ func (r *resourceReconciler) reconcile(req ctrlrt.Request) error {
 		r.cfg, r.log, r.metrics, r, sess, acctID, region,
 	)
 	if err != nil {
-		return err
+		return ctrlrt.Result{}, err
 	}
+	return r.handleReconcileError(rlog, r.reconcile(ctx, rlog, rm, res))
+}
 
+func (r *resourceReconciler) reconcile(
+	ctx context.Context,
+	rlog acktypes.AWSLogger,
+	rm acktypes.AWSResourceManager,
+	res acktypes.AWSResource,
+) error {
 	if res.IsBeingDeleted() {
-		return r.cleanup(ctx, rm, res)
+		return r.cleanup(ctx, rlog, rm, res)
```

Comment:
Not a big deal, but you can also pass the logger via the context. `ctx.WithValue("logger", rlog)` https://golang.org/pkg/context/#WithValue

Reply:
good idea.